src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,48 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a person's contact information in the address book
+ */
+public class Contact {
+    public String value;
+    protected boolean isPrivate;
+
+    public Contact(String value, boolean isPrivate, String validationRegex, String constraints) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        String trimmedValue = value.trim();
+        if (!isValid(trimmedValue, validationRegex)) {
+            throw new IllegalValueException(constraints);
+        }
+        this.value = trimmedValue;
+    }
+
+    /**
+     * Checks if value matches the regex for validation
+     */
+    public static boolean isValid(String test, String validationRegex) {
+        return test.matches(validationRegex);
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                        && this.value.equals(((Contact) other).value)); // state check
+    }
+
+    @Override
+    public int hashCode() {
+        return value.hashCode();
+    }
+
+    public boolean isPrivate() {
```

Comment:
Do refer back to coding standard on commenting.
Public trivial methods must still be commented.